src/signers/private-key-signer-btc.js

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+// Copyright 2024 Tether Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { networks, Psbt } from 'bitcoinjs-lib'
+import * as ecc from '@bitcoinerlab/secp256k1'
+import { ECPairFactory } from 'ecpair'
+
+// eslint-disable-next-line camelcase
+import { sodium_memzero } from 'sodium-universal'
+
+import { hashMessage, buildPaymentScript, detectInputOwnership, ensureWitnessUtxoIfNeeded, normalizeConfig, getAddressFromPublicKey } from './utils.js'
+
+const ECPair = ECPairFactory(ecc)
+/** @typedef {import('../wallet-account-read-only-btc.js').BtcWalletConfig} BtcWalletConfig */
+
+/**
+ * Signer backed by a single raw private key (non-HD).
+ * - Does not support HD derivation, extended keys, or master fingerprint.
+ * - Signs messages and PSBTs directly using the leaf key.
+ */
+export default class PrivateKeySignerBtc {
+  /**
+   * @param {string | Uint8Array | Buffer} privateKey - Hex string or bytes of the raw private key
+   * @param {BtcWalletConfig} [config] - Network/BIP configuration (bip defaults to 44, network to 'bitcoin')
+   */
+  constructor (privateKey, config = {}) {
+    const cfg = normalizeConfig(config)
+    const pkBuf = typeof privateKey === 'string'
+      ? Buffer.from(privateKey, 'hex')
+      : Buffer.from(privateKey)
```

Comment:
if privateKey is a Buffer or Uint8Array, why you are creating a new Buffer copying it?